service_discovery/core/src/main.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+//! Project Eclipse Chariott Service Discovery
+//!
+//! This is the Service Discovery system for Chariott. It includes a service registry,
+//! which is a database of services that are currently registered. Other applications
+//! can find the metadata for registered services.
+
+// Tells cargo to warn if a doc comment is missing and should be provided.
+#![warn(missing_docs)]
+
+use parking_lot::RwLock;
+use proto_servicediscovery::chariott_registry::v1::registry_server::RegistryServer;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tonic::transport::Server;
+use tracing::{debug, info};
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::EnvFilter;
+
+mod registry_impl;
+
+/// Endpoint for the chariott service registry
+const CHARIOTT_SERVICE_REGISTRY_ADDR: &str = "0.0.0.0:50000";
```

Comment:
chariott is the code name; what about leaving it out of the code itself and just name each class based on its functionality

Reply:
I have removed it from all new code, kept it in the existing package name, as well as comments describing the files. Please let me know what you think

Reply:
we can look at the package names later; maybe open a tracking issue so we do not forget about it